enterprise/frontend/src/metabase-enterprise/caching/components/DashboardCacheSection/DashboardCacheSection.tsx

```diff
@@ -1,18 +1,53 @@
-import type { Dashboard } from "metabase-types/api";
+import { useMemo } from "react";
+import { t } from "ttag";
+import _ from "underscore";
 
-import CacheSection from "../CacheSection";
+import { useCacheConfigs } from "metabase/admin/performance/hooks/useCacheConfigs";
+import { getShortStrategyLabel } from "metabase/admin/performance/strategies";
+import { DelayedLoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper/DelayedLoadingAndErrorWrapper";
+import type { DashboardSidebarPageProps } from "metabase/dashboard/components/DashboardInfoSidebar";
+import { Flex } from "metabase/ui";
 
-interface DashboardCacheSectionProps {
-  dashboard: Dashboard;
-  onSave: (cache_ttl: number | null) => Promise<Dashboard>;
-}
+import { CacheSectionRoot } from "../CacheSection/CacheSection.styled";
+import { PolicyToken } from "../StrategyFormLauncher.styled";
+import { getDashboardId } from "../utils";
 
-const DashboardCacheSection = ({
+export const DashboardCacheSection = ({
   dashboard,
-  onSave,
-}: DashboardCacheSectionProps) => {
-  return <CacheSection initialCacheTTL={dashboard.cache_ttl} onSave={onSave} />;
-};
+  setPage,
+}: DashboardSidebarPageProps) => {
+  const dashboardId = getDashboardId(dashboard);
+
+  const { configs, loading, error } = useCacheConfigs({
+    configurableModels: ["dashboard"],
+    id: dashboardId,
+  });
+
+  const targetConfig = useMemo(
+    () => _.findWhere(configs, { model_id: dashboardId }),
+    [configs, dashboardId],
+  );
+  const savedStrategy = targetConfig?.strategy;
 
-// eslint-disable-next-line import/no-default-export -- deprecated usage
-export default DashboardCacheSection;
+  const shortStrategyLabel =
+    getShortStrategyLabel(savedStrategy, "dashboard") || t`Use default`;
+
+  return (
+    <DelayedLoadingAndErrorWrapper loading={loading} error={error}>
+      <CacheSectionRoot>
+        <Flex align="center" justify="space-between">
+          {t`Caching policy`}
+          <PolicyToken
+            onClick={() => setPage("caching")}
+            variant="subtle"
+            radius={0}
+            p={0}
+            style={{ border: "none" }}
+          >
+            {shortStrategyLabel}
+          </PolicyToken>
+        </Flex>
+      </CacheSectionRoot>
+    </DelayedLoadingAndErrorWrapper>
```

Comment:
Could you rework this loading state? I think it should show some sort of loading indicator while it's loading the strategy rather than just keeping it blank. It's a confusing experience when loading a larger dashboard  to just have it not there.

![loadingstate](https://github.com/metabase/metabase/assets/30528226/f4f44657-fa01-4d64-a492-e1d7e97896da)

Reply:
This isn't the intended behavior. Looking into it

Reply:
I'm still seeing this on larger dashboards. doesn't need to block, but it's not ideal, especially because larger dashboards are the ones where caching is most valuable

Reply:
Thanks for catching this. I wasn't reproducing it (even with throttling), I guess because I don't have such a sumptuous dashboard in my local instance 😁 . Will look into it